pyaerocom/io/test/test_readgridded.py

```diff
@@ -57,34 +58,120 @@ def test_ReadGridded_class_empty():
     assert r.data_dir == None
     from pyaerocom.io.aerocom_browser import AerocomBrowser
     assert isinstance(r.browser, AerocomBrowser)
-
-    failed = False
-    try:
+    with pytest.raises(AttributeError):
         r.years_avail
-    except AttributeError:
-        failed = True
-    assert failed
     assert r.vars_filename == []
 
+
+@testdata_unavail
+def test_ReadGridded_data_dir(reader_tm5):
+    assert reader_tm5.data_dir == path_tm5
+    assert reader_tm5._vars_2d == ['abs550aer', 'od550aer']
+    assert reader_tm5._vars_3d == []
+
+
+def test_ReadGridded_ts_types():
+    r = ReadGridded(data_dir=path_tm5)
+    assert sorted(r.ts_types) == ['daily', 'monthly']
+
+@testdata_unavail
+def test_ReadGridded_read_var(reader_tm5):
+    r = reader_tm5
+    data = r.read_var('od550aer')
+    assert isinstance(data, GriddedData)
+    ds = data.to_xarray().load()
+    mean = float(ds.mean(dim='time').mean(dim='lat').mean(dim='lon'))
+    npt.assert_almost_equal(mean, 0.09631748)
+    with pytest.raises(VarNotAvailableError):
+        r.read_var('wetso4')
+    from pyaerocom.io.aux_read_cubes import add_cubes
+    gridded = r.read_var('new_var', aux_vars=[
+        'abs550aer', 'od550aer'], aux_fun=add_cubes)
+    assert gridded.var_name == 'new_var'
+    assert isinstance(gridded, GriddedData)
+
+
+@pytest.mark.parametrize('experiments', [
+    (['exp1']),
+    (['exp1', 'exp2']),
+])
+def test_ReadGridded_experiments(tmpdir, experiments):
+    for exp in experiments:
+        filename = 'aerocom3_TM5-met2010_{}-CTRL2019_abs550aer_Column_2010_daily.nc'.format(exp)
+        open(os.path.join(tmpdir, filename), 'a').close()
+    r = ReadGridded(data_dir = str(tmpdir))
+    assert r.experiments == experiments
+
+@pytest.mark.parametrize('vars,expected', [
+    (['mmro3', 'rho'], ['conco3', 'mmro3', 'rho']),
+    (['od440aer', 'od870aer'], ['ang4487aer', 'od440aer', 'od870aer']),
+
+])
+def test_ReadGridded_aux(tmpdir, vars, expected):
+    for var in vars:
+        filename = 'aerocom3_TM5-met2010_AP3-CTRL2019_{}_Column_2010_daily.nc'.format(
+            var)
+        open(os.path.join(tmpdir, filename), 'a').close()
+    r = ReadGridded(data_dir=str(tmpdir))
+    for var in expected:
+        assert r.has_var(var)  # calling has_var
+    assert sorted(r.vars_provided) == sorted(expected)
+    # assert r.has_var('conco3')
+
+
+def test_ReadGridded_prefer_longer():
+    r = ReadGridded(data_dir=path_tm5)
+    daily = r.read_var('abs550aer', ts_type='monthly', flex_ts_type=True,
+                       prefer_longer=True)
+    assert daily.ts_type == 'daily'
```

Comment:
Is there also a monthly field? Consider using parametrize with different settings?

Reply:
Yes, abs550aer has daily and monthly files in the testdata.
Running this with ts_type='yearly' and flex_ts_type=True results in `pyaerocom.exceptions.DataQueryError: Could not find file match for query`. Shouldn't flex_ts_type option ensure that it chooses the monthly or daily files?

Reply:
Seems that the current behavior is that setting `ts_type` leads to `flex_ts_type` not affecting the behavior.

Reply:
Hmm, okay, can you make an issue for that?

Reply:
Yep!